### PR TITLE
Remove many instances of "py3 review required" comments

### LINKED ISSUE #9888: Threshold_py3_staging: look at "Py3 rview" comments in the code, evaluate the code and remove these comments
During the Python 3 migration, several extra comments have been added to clarify changes. Now we know more about how Python 3 behaves and we are able to test things with a working branch, we can re-evaluate these cases and remove the comments.

For example, there are several cases of `for value in list(dictionary.values())`, in which case wrapping inside a list call is not necessary. In case of `for key in list(dictionary.keys())`, it is enough to use `for key in dictionary:`

Some other comments are obsolete in that they tell the reader that we are iterating over the keys of a dictionary, but if we call .keys() on an object, it is pretty obvious that we're dealing with a dictionary anyway.

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Closes #9888 

### Summary of the issue:
During the Python 3 transition, many advises for additional review have been added to the code, particularly in the case of dict iterations.

### Description of how this pull request fixes the issue:
 I've gone through most of these now and removed them, performed appropriate actions. Most of these were cases of:

* A call was wrapped in a list unnecessarily
* Iterating over dictionary.keys() while iterating over dictionary gives us the keys anyway.

### Testing performed:
Tested that NVDA still works as expected from source.

### Known issues with pull request:
Two py3 review statements i added to NVDAObjects/window/edit are still there as they deal with a seemingly off by one error that has been there for ages and requires more investigation.

### Change log entry:
None

### CODE REVIEW

**Review by josephsl — APPROVED** [review]
Reviewed while there is no earthquake...

**Comment by Brian1Gaff** [comment]
Is this the one where you try to count list items if there are no data for 
the number of items. Ie the guess option in the dialogue.  I see this in 
tabs and odd lists where you tend to get 1 more number than there actually 
are in the list.


bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users

**Review by feerrenrut — COMMENTED** [review]
I've looked through this all, just a question about one of the type hints.

File: source/config/__init__.py [hunk 1]
```diff
@@ -369,28 +370,27 @@ class ConfigManager(object):
 	def __init__(self):
 		self.spec = confspec
 		#: All loaded profiles by name.
-		self._profileCache = {}
+		self._profileCache: Optional[Dict[Optional[str], ConfigObj]] = {}
```
[the remark this answers is not in the document]
Comment (leonardder) [reply source/config/__init__.py]: None is the key for the master profile.

**Pull request merged**